src/libsystemd/sd-bus/bus-objects.c

```diff
@@ -1610,6 +1610,39 @@ static int vtable_member_compare_func(const struct vtable_member *x, const struc
 
 DEFINE_PRIVATE_HASH_OPS(vtable_member_hash_ops, struct vtable_member, vtable_member_hash_func, vtable_member_compare_func);
 
+static bool names_are_valid(const char *signature, const char **names, int *flags) {
+        enum {
+                first           = 1<<0,
+                have_names      = 1<<1,
+                alone           = 1<<2,
+        };
+
+        if((*flags&first || *flags&alone ) && **names)
+                *flags |= have_names;
+
+        for (;;) {
+                size_t l;
+
+                if (!*signature)
+                        break;
+
+                signature_element_length(signature, &l);
+
+                if (**names) {
+                        if(!member_name_is_valid(*names))
+                                return false;
+                        *names += strlen(*names) + 1;
+                } else if(*flags&have_names)
+                        return false;
+
+                signature += l;
+        }
+        if(*flags&have_names && **names && !*flags&first)
```

Comment:
whitespace totally off again

Reply:
also, the expression looks wrong to me. I am pretty sure `!` binds before `&`, no? hence the `!*flags&first` expression not going to do what you think it does, no?

Reply:
please add a comment here, `/* let's check if there are more argument names specified than the signature allows */` or something along those lines